corehq/apps/hqwebapp/middleware.py

```diff
@@ -27,11 +24,10 @@ def _reject(self, request, reason):
         _assert = soft_assert('{}@{}'.format('sreddy+logs', 'dimagi.com'), exponential_backoff=False)
         _assert(False, warning)
 
-        if '/login/' in request_url:
-            # exempt login URLs from csrf, but still get failure logs
+        if ('/login/' in request_url and
+           settings.CSRF_SOFT_MODE and
+           reason in [REASON_NO_CSRF_COOKIE, REASON_BAD_TOKEN]):
+            # exempt login URLs from csrf if CSRF_SOFT_MODE is on
             self._accept(request)
```

Comment:
That's true if the returned value is what actually causes acceptance or rejection. On the other hand, if `_accept()` is setting a response header and `_reject()` does nothing, then it might not matter what is returned. I haven't looked at the source code, so I'm not sure.